cmd/snap/cmd_pack.go

```diff
@@ -25,27 +25,41 @@ import (
 	"github.com/jessevdk/go-flags"
 
 	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/snap"
 	"github.com/snapcore/snapd/snap/pack"
 )
 
 type packCmd struct {
-	Positional struct {
+	CheckSkeleton bool `long:"check-skeleton"`
+	Positional    struct {
 		SnapDir   string `positional-arg-name:"<snap-dir>"`
 		TargetDir string `positional-arg-name:"<target-dir>"`
 	} `positional-args:"yes"`
 }
 
-var shortPackHelp = i18n.G("Pack the given target dir as a snap")
+var shortPackHelp = i18n.G("Pack the given directory as a snap")
 var longPackHelp = i18n.G(`
-The pack command packs the given snap-dir as a snap.`)
+The pack command packs the given snap-dir as a snap and writes the result to
+target-dir. If target-dir is omitted, the result is written to current
+directory. If both source-dir and target-dir are omitted, the pack command packs
+the current directory.
+
+When used with --check-skeleton, pack only checks whether snap-dir contains
+valid snap metadata and raises an error otherwise. Application commands listed
+in snap metadata file, but appearing with incorrect permission bits or missing
+from snap-dir are signaled through diagnostic messages and do not result in an
+error.
```

Comment:
Application commands listed in the snap metadata but _missing_ won't result in an error, but incorrect permission bits will.

Reply:
Silly me, thanks for catching this.